mantidimaging/gui/windows/spectrum_viewer/model.py

```diff
@@ -349,20 +353,22 @@ def save_csv(self, path: Path, normalise: bool, normalise_with_shuttercount: boo
             csv_output.add_column("ToF", self.units.tof_seconds_to_us(), "Microseconds")
             csv_output.add_column("Energy", self.units.tof_seconds_to_energy(), "MeV")
 
-        for roi_name in self.get_list_of_roi_names():
-            csv_output.add_column(roi_name, self.get_spectrum(roi_name, SpecType.SAMPLE, normalise_with_shuttercount),
-                                  "Counts")
+        for roi_name, roi in rois.items():
+            sample_spectrum = self.get_spectrum(roi, SpecType.SAMPLE, normalise_with_shuttercount)
+            csv_output.add_column(roi_name, sample_spectrum, "Counts")
+
             if normalise:
                 if self._normalise_stack is None:
                     raise RuntimeError("No normalisation stack selected")
-                csv_output.add_column(roi_name + "_open", self.get_spectrum(roi_name, SpecType.OPEN), "Counts")
-                csv_output.add_column(roi_name + "_norm",
-                                      self.get_spectrum(roi_name, SpecType.SAMPLE_NORMED, normalise_with_shuttercount),
-                                      "Counts")
 
-        with path.open("w") as outfile:
+                open_spectrum = self.get_spectrum(roi, SpecType.OPEN)
+                norm_spectrum = self.get_spectrum(roi, SpecType.SAMPLE_NORMED, normalise_with_shuttercount)
+
+                csv_output.add_column(f"{roi_name}_open", open_spectrum, "Counts")
+                csv_output.add_column(f"{roi_name}_norm", norm_spectrum, "Counts")
+
+        with path.open("w", encoding="utf-8") as outfile:
```

Comment:
No need to add `encoding="utf-8"` here, will save needing to change it in the test as well.